tuxemon/event/actions/get_player_monster.py

```diff
@@ -20,33 +29,183 @@
+                and target.has_type(ElementType(value_name))
+            ):
+                self.result = True
+                return True
+            # filter shape
+            if (
+                filter_name == "shape"
+                and value_name in list(MonsterShape)
+                and target.shape == value_name
+            ):
+                self.result = True
+                return True
+            # filter taste warm
+            if (
+                filter_name == "taste_warm"
+                and value_name in list(TasteWarm)
+                and target.taste_warm == value_name
+            ):
+                self.result = True
+                return True
+            # filter taste cold
+            if (
+                filter_name == "taste_cold"
+                and value_name in list(TasteCold)
+                and target.taste_cold == value_name
+            ):
+                self.result = True
+                return True
+
+            # filter numeric fields
+            if self.extra is not None:
+                field = 0
+                if filter_name == "level":
+                    field = target.level
+                elif filter_name == "weight":
+                    field = int(target.weight)
+                elif filter_name == "height":
+                    field = int(target.height)
+                elif filter_name == "max_hp":
+                    field = target.hp
+                elif filter_name == "current_hp":
+                    field = target.current_hp
+                elif filter_name == "armour":
+                    field = target.armour
+                elif filter_name == "dodge":
+                    field = target.dodge
+                elif filter_name == "melee":
+                    field = target.melee
+                elif filter_name == "ranged":
+                    field = target.ranged
+                elif filter_name == "speed":
+                    field = target.speed
+                extra = int(self.extra)
+                if value_name == "less_than" and bool(lt(field, extra)):
+                    self.result = True
+                    return True
+                elif value_name == "less_or_equal" and bool(le(field, extra)):
+                    self.result = True
+                    return True
+                elif value_name == "greater_than" and bool(gt(field, extra)):
+                    self.result = True
+                    return True
+                elif value_name == "greater_or_equal" and bool(
+                    ge(field, extra)
+                ):
+                    self.result = True
+                    return True
+                elif value_name == "equals" and bool(eq(field, extra)):
+                    self.result = True
+                    return True
+                elif value_name == "not_equals" and bool(ne(field, extra)):
+                    self.result = True
+                    return True
+
+        return False
```

Comment:
Shouldn't be `self.result` set to False in the default case as well?
Or at the beginning of the method maybe.

Also, I think we could just return the value of `self.result` in every cases, instead of repeating the same boolean value.

Reply:
You're right, I'll look carefully

Reply:
done, it returns `self.result`
`self.default` is set to False in the Start and it propagates.